src/mumble/AudioOutput.cpp

```diff
@@ -100,27 +100,37 @@ AudioOutput::~AudioOutput() {
 float AudioOutput::calcGain(float dotproduct, float distance) {
 	// dotproduct is in the range [-1, 1], thus we renormalize it to the range [0, 1]
 	float dotfactor = (dotproduct + 1.0f) / 2.0f;
-	float att;
 
+	// Volume on the ear opposite to the sound should never reach 0 in the real world.
+	// Therefore, we define the minimum volume as 1/4th of the theoretical maximum (ignoring
+	// the sound direction but taking distance into account) for _any_ ear.
+	const float offset = (1.0f - dotfactor) * 0.25f;
+	dotfactor += offset;
+
+	float att;
 
-	// No distance attenuation
-	if (Global::get().s.fAudioMaxDistVolume > 0.99f) {
-		att = qMin(1.0f, dotfactor + Global::get().s.fAudioBloom);
+	if (distance < 0.0001f) {
+		// Listener is inside source -> no attenuation
+		att = 1.0f;
```

Comment:
Is this only to save unnecessary computation or has this extra branch a more subtle purpose?

If it is only about saving computation:
This equates to a distance of 0.0001m, which is 0.01cm. I think we can default to no attenuation way earlier. I would suggest 0.1m -> `0.1f`.
Though in the end it would seem fine to me to just let the code fall-through to the blooming-branch below :eyes:

Reply:
This branch covers https://github.com/mumble-voip/mumble/pull/6234#issuecomment-1757613551

But we might get away with a larger value, yes.

Reply:
As I said in the earlier comment, I assume this is because there is a inherent distance between the speakers of the player and the player position thus the distance will be close, but never exactly 0. And therefore, the volume will never be exactly 1.

Reply:
But for distances small enough that this separate branch is taken, the fading function should be so close to 1 that we don't end up with an audible volume reduction. And given that we in fact _do_ have such an audible reduction, I deduce that the fading function doesn't work the way we would expect it to...
Anyway, I'm happy with keeping the current approach. If someone fancies, the fading function problem can be dealt with in a separate PR.

Reply:
I see what you mean. So the ``bloomfac`` in line 121 can have a maximum of ``Global::get().s.fAudioBloom``, which is capped to 75% or something.
And then ``dotfactor`` may have a value of 0.5, because the dotproduct is probably 0. (since we have a directional zero vector). I do not see how 0.75 + 0.5 should be less than 1 though :thinking: 

Annnnyway, I think we should keep this branch in. Especially now that the check distance is larger.

Reply:
It might be possible that I originally tried this with bloom 0. Which - still- should not decrease the volume when we are literally _inside_ the source.

Reply:
Right, that's actually it! If ``bloom < 0.5``, we will get a ``volume < 1`` for ``dist == 0``, but as soon as we are not numerically inside the source, one of the channels will have a dotfactor of ``1``. That is the problem - a numerical error at exactly ``dist == 0`` so the extra check for that case is probably warranted.

Reply:
I think we could also get rid of the blooming now that we have the minimum volume on all speakers. I think blooming was effectively mimicking that. Anyway, this need not be done in this PR (if at all) :)

Reply:
I think the bloom is meant to be a different effect. We should keep it as it produces significant changes to the perceived audio at close range to the source. The tiny minimum volume is more of a psycho-acoustic thing I suppose. I am not an audiophile though :upside_down_face: